digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/ChangeLanesAccordingToVvhChanges.scala

```diff
@@ -0,0 +1,147 @@
+    val groupedAssets = allLanes.groupBy(_.linkId)
+    val (changedLanes, changeSet) = laneFiller.fillTopology(roadLinks, groupedAssets, Some(changedSet))
+
+    val generatedMappedById = changeSet.generatedPersistedLanes.groupBy(_.id)
+    val modifiedLanes = projectedLanes.filterNot(lane => generatedMappedById(lane.id).nonEmpty) ++ changeSet.generatedPersistedLanes
+
+    updateChangeSet(changeSet)
+    persistModifiedLinearAssets(modifiedLanes)
+    changedLanes
+  }
+
+  def updateChangeSet(changeSet: ChangeSet) : Unit = {
+    def treatChangeSetData(changeSetToTreat: Seq[baseAdjustment]): Unit = {
+      val toAdjustLanes = getPersistedLanesByIds(changeSetToTreat.map(_.laneId).toSet, false)
+
+      changeSetToTreat.foreach { adjustment =>
+        val oldLane = toAdjustLanes.find(_.id == adjustment.laneId).get
+        val newLane = persistedToNewLaneWithNewMeasures(oldLane, adjustment.startMeasure, adjustment.endMeasure)
+        val newLaneID = create(Seq(newLane), Set(oldLane.linkId), oldLane.sideCode, VvhGenerated)
+        moveToHistory(oldLane.id, Some(newLaneID.head), true, true, VvhGenerated)
+      }
+    }
+
+    def persistedToNewLaneWithNewMeasures(persistedLane: PersistedLane, newStartMeasure: Double, newEndMeasure: Double): NewLane = {
+      NewLane(0, newStartMeasure, newEndMeasure, persistedLane.municipalityCode, false, false, persistedLane.attributes)
+    }
+
+    PostGISDatabase.withDynTransaction {
+      if (changeSet.adjustedSideCodes.nonEmpty)
+        logger.info("Saving SideCode adjustments for lane/link ids=" + changeSet.adjustedSideCodes.map(a => "" + a.laneId).mkString(", "))
+
+      changeSet.adjustedSideCodes.foreach { adjustment =>
+        moveToHistory(adjustment.laneId, None, false, false, VvhGenerated)
+        dao.updateSideCode(adjustment.laneId, adjustment.sideCode.value, VvhGenerated)
+      }
+
+      if (changeSet.adjustedMValues.nonEmpty)
+        logger.info("Saving adjustments for lane/link ids=" + changeSet.adjustedMValues.map(a => "" + a.laneId + "/" + a.linkId).mkString(", "))
+
+      treatChangeSetData(changeSet.adjustedMValues)
+
+      if (changeSet.adjustedVVHChanges.nonEmpty)
+        logger.info("Saving adjustments for lane/link ids=" + changeSet.adjustedVVHChanges.map(a => "" + a.laneId + "/" + a.linkId).mkString(", "))
+
+      treatChangeSetData(changeSet.adjustedVVHChanges)
+
+      val ids = changeSet.expiredLaneIds.toSeq
+      if (ids.nonEmpty)
+        logger.info("Expiring ids " + ids.mkString(", "))
+      ids.foreach(moveToHistory(_, None, true, true, VvhGenerated))
+    }
+  }
+
+  def persistModifiedLinearAssets(newLanes: Seq[PersistedLane]): Unit = {
+    if (newLanes.nonEmpty) {
+      logger.info("Saving modified lanes")
+
+      val username = "modifiedLanes"
+      val (toInsert, toUpdate) = newLanes.partition(_.id == 0L)
+
+      PostGISDatabase.withDynTransaction {
```

Comment:
nää tapahtuu nyt kahdessa erillisessä tranastiossa, voidaan varmaan alkuun mennä näin ja katsoa miten toimii. 
Tuli mieleen että vois tranaction per linkki. Silloin onnistuneet linkit tallennetaan kantaan ja epäonnistuneet rollback, nyt tää taitaa rollback kaikki annetut linkit